ydb/core/cms/sentinel.cpp

```diff
@@ -41,6 +43,41 @@ namespace NKikimr::NCms {
 
 namespace NSentinel {
 
+/// TNodeStatusComputer
+
+ui32 TNodeStatusComputer::GetCurrentNodeState() const {
+    return (ui32)ActualState;
+}
+
+bool TNodeStatusComputer::Compute() {
+    if (ActualState != CurrentState
+        && ((CurrentState == ENodeState::BAD && StateCounter >= BadStateLimit)
+        || (CurrentState == ENodeState::GOOD && StateCounter >= GoodStateLimit))) {
+        ActualState = CurrentState;
+        return true;
+    }
+    if (ActualState != ENodeState::MAY_BE_BAD && (CurrentState == ENodeState::BAD && StateCounter < BadStateLimit)) {
+        ActualState = ENodeState::MAY_BE_BAD;
+        return false;
+    }
+    if (ActualState != ENodeState::MAY_BE_GOOD && (CurrentState == ENodeState::GOOD && StateCounter < GoodStateLimit)) {
+        ActualState = ENodeState::MAY_BE_GOOD;
+        return false;
+    }
+    return false;
```

Comment:
```suggestion
    if (ActualState != CurrentState && (DefinitelyBad() || DefinitelyGood())) {
        ActualState = CurrentState;
        return true;
    }
    if (ActualState != ENodeState::MAY_BE_BAD && MaybeBad()) {
        ActualState = ENodeState::MAY_BE_BAD;
        return false;
    }
    if (ActualState != ENodeState::MAY_BE_GOOD && MaybeGood()) {
        ActualState = ENodeState::MAY_BE_GOOD;
        return false;
    }
    return false;
```

И заинлайнить вот эти вспомогательные функции:
```cpp
bool TNodeStatusComputer::MaybeBad() const {
    return CurrentState == ENodeState::BAD && StateCounter < BadStateLimit;
}

bool TNodeStatusComputer::DefinitelyBad() const {
    return CurrentState == ENodeState::BAD && StateCounter >= BadStateLimit;
}

bool TNodeStatusComputer::MaybeGood() const {
    return CurrentState == ENodeState::GOOD && StateCounter < GoodStateLimit;
}

bool TNodeStatusComputer::DefinitelyGood() const {
    return CurrentState == ENodeState::GOOD && StateCounter >= GoodStateLimit;
}
```

По крайней мере мне так было бы сильно понятнее.